### ISSUE TITLE
Question - what docker container works for a MVC application after big rename

### ISSUE THREAD

**Issue by jsacapdev** [aspnet/aspnet-docker#44]
So, after the rename, I have been trying to get a very simple mvc application working in a docker container. Here is my project.json:

{
    "dependencies": {
        "Microsoft.AspNet.Server.IIS": "1.0.0-*",
        "Microsoft.AspNet.Diagnostics": "1.0.0-*",
        "Microsoft.AspNet.Mvc": "6.0.0-*",
        "Microsoft.Framework.Logging.Console": "1.0.0-*",
        "Microsoft.Framework.ConfigurationModel.Json": "1.0.0-*",
        "Kestrel": "1.0.0-*"
    },
    "commands": {
        "web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5001",
        "kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5004"
    },
	 "frameworks": {
        "dnx451": {},
        "dnxcore50": {}
  }
}

I built a docker image using the following:

FROM microsoft/aspnet:nightly

COPY . /app

WORKDIR /app

RUN ["kpm", "restore"]

EXPOSE 5004

ENTRYPOINT ["dnx", ".", "kestrel"]

When built, here's the dnvm list:

root@a9b8056a2ee4:/app# dnvm list

Active Version              Runtime Location             Alias
------ -------              ------- --------             -----
  *    1.0.0-beta4-11351    mono    /opt/dnx/runtimes    default

I am able to restore and execute (dnx .) the application on my local (windows sever) development platform. But in a docker container, I cannot get it to run. It fails, with the same error reported in:

https://github.com/aspnet/DataProtection/issues/70

This issue is marked as a duplicate of:

https://github.com/aspnet/DataProtection/issues/68

And (even though it is a bit unclear), this issue suggests that the issue may be fixed in mono 4.1 (ish).

The docker nightly uses a mono 3.12 build. If this issue is fixed in mono 4+, does that mean that for any software developed based on the big rename, a new docker container will be required? 

Or is it the case that this issue is independent of the docker image itself. It all depends on when the fixes are put in place that the docker image will pull in when when it is being built.

Either way - can you let me know when a image will be available that I cant use to run the asp.net 5 samples I am creating?

**Comment by ahmetalpbalkan** [comment]
@jsacapdev we will be publishing the beta4 image as soon as we can once it goes out (https://github.com/aspnet/Home/releases) and we will be updating to mono 4.x (it's not also out yet. https://github.com/mono/mono/releases)

cc: @danroth27

**Comment by ryan1234** [comment]
It was helpful for me to get unblocked while I wait for beta4 to build my own mono 4.x Docker image.

```
FROM mono:3.12

RUN apt-get -qq update && apt-get -qqy install unzip
RUN apt-get -qq update && apt-get install -qqy procps && apt-get -qqy install git

# Upgrade mono to latest on master. Needed until next release
RUN apt-get -qq update \
    && apt-get install -qqy gettext autoconf libtool automake build-essential \
    && PATH=/mono/bin:$PATH \
    && cd /root \
    && git clone https://github.com/mono/mono.git \
    && cd mono \
    && ./autogen.sh --prefix=/mono \
    && make \
    && make install
    
# Otherwise it will use old mono (search /mono/bin first!)
ENV PATH /mono/bin:$PATH
```
Not hard to build at all. Then I based a vNext image off of mono 4.x instead of mono 3. 

Hope that helps a little.

**Comment by ahmetalpbalkan** [comment]
@ryan1234 thanks for sharing your solution.

**Comment by Jarga** [comment]
I had the same solution as above, to build my own image with the latest from the mono repo. Though i noticed that the image size gets pretty big if you don't do some cleanup of the files pulled down for the git clone.

```

FROM mono:3.12

MAINTAINER Sean McAdams <sean.j.mcadams@gmail.com>

#get build dependencies and download mono
RUN apt-get update -y \
    && apt-get install -y git autoconf libtool automake build-essential mono-devel gettext \
    && PATH=/usr/local/bin:$PATH \
    && git clone https://github.com/mono/mono.git

#Move to mono directory for build process
WORKDIR mono

RUN ./autogen.sh --prefix="/usr/local" \
    && make \
    && make install \
    && rm /mono -r
```

I also have a pre-build image i stored on docker as ```jarga/mono-4.1.0```

Obviously the official version will be more optimized and smaller, but it can get some people un-stuck

**Comment by jsacapdev** [comment]
@ahmetalpbalkan - thanks for the answer. @ryan1234 and @Jarga - thanks for the guidance.

**Comment by AlexZeitler** [comment]
I wrote a blog post about running beta 4 with DNX in Docker:
https://alexanderzeitler.com/articles/Running-ASP.NET-5-beta4-in-Docker-with-DNX-runtime/
